File: python/Neetcode/LinkedList/Easy/141.LinkedListCycle.py
# Given head, the head of a linked list, determine if the linked list has a cycle.

class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

def hasCycle(head):
    slow, fast = head, head

    while fast and fast.next:
        slow = slow.next
        fast = fast.next.next

        if slow == fast:
            return True

    return False

File: swift/Neetcode/LinkedList/Easy/141.LinkedListCycle.swift
// Given head, the head of a linked list, determine if the linked list has a cycle.

func hasCycle(_ head: ListNode?) -> Bool {
    var slow = head, right = head

    while let next = right?.next {
        slow = slow?.next
        right = next.next
        
        if slow === right { return true }
    }

    return false
}